packages/vite/src/node/plugins/wasm.ts

```diff
@@ -7,11 +7,19 @@ const wasmHelperId = '/__vite-wasm-helper'
 const wasmHelper = async (opts = {}, url: string) => {
   let result
   if (url.startsWith('data:')) {
-    // @ts-ignore
-    const binaryString = atob(url.replace(/^data:.*?base64,/, ''))
-    const bytes = new Uint8Array(binaryString.length)
-    for (let i = 0; i < binaryString.length; i++) {
-      bytes[i] = binaryString.charCodeAt(i)
+    const urlContent = url.replace(/^data:.*?base64,/, '')
+    let bytes
+    if (typeof Buffer === 'function' && typeof Buffer.from === 'function') {
+      bytes = Buffer.from(urlContent, 'base64')
+    } else if (typeof atob === 'function') {
+      // @ts-ignore
+      const binaryString = atob(urlContent)
+      const bytes = new Uint8Array(binaryString.length)
+      for (let i = 0; i < binaryString.length; i++) {
+        bytes[i] = binaryString.charCodeAt(i)
+      }
```

Comment:
The for loop is assigning to the local `bytes` variable, but we need to assign to the outer `bytes` variable instead